pkgs/applications/office/libreoffice/wrapper.nix

```diff
@@ -1,33 +1,101 @@
-{ lib, runCommand
-, libreoffice, dbus, bash, substituteAll
-, coreutils, gnugrep
-, dolphinTemplates ? true
+{ lib
+, stdenv
+# The unwrapped libreoffice derivation
+, unwrapped
+, makeWrapper
+, runCommand
+, substituteAll
+# For Emulating wrapGAppsHook
+, gsettings-desktop-schemas
+, hicolor-icon-theme
+, dconf
+, librsvg
+, gdk-pixbuf
+# Configuration options for the wrapper
+, extraMakeWrapperArgs ? []
+, dbusVerify ? stdenv.isLinux
+, dbus
 }:
-runCommand libreoffice.name {
-  inherit (libreoffice) jdk meta;
-  inherit coreutils dbus gnugrep libreoffice bash;
+
+let
+  makeWrapperArgs = builtins.concatStringsSep " " ([
+    "--set" "GDK_PIXBUF_MODULE_FILE" "${librsvg}/${gdk-pixbuf.moduleDir}.cache"
+    "--prefix" "GIO_EXTRA_MODULES" ":" "${lib.getLib dconf}/lib/gio/modules"
+    "--prefix" "XDG_DATA_DIRS" ":" "${unwrapped.gtk3}/share/gsettings-schemas/${unwrapped.gtk3.name}"
+    "--prefix" "XDG_DATA_DIRS" ":" "$out/share"
+    "--prefix" "XDG_DATA_DIRS" ":" "${gsettings-desktop-schemas}/share/gsettings-schemas/${gsettings-desktop-schemas.name}"
+    "--prefix" "XDG_DATA_DIRS" ":" "${hicolor-icon-theme}/share"
```

Comment:
Does makeWrapper generate less shell if we combine this?

Reply:
> Does makeWrapper generate less shell if we combine this?

Yes. But on the other hand, reading the generated wrapper is easier IMO when each environment variable has it's own 5 lines of shell variables assignments. If the maintainer prefers these environment variables combined, I won't mind change it.